ext/symengine/symengine_utils.c

```diff
@@ -59,10 +61,16 @@ void sympify(VALUE operand2, basic_struct *cbasic_operand2) {
             break;
 
         case T_DATA:
+            c = rb_obj_classname(operand2);
+            if(strcmp(c, "BigDecimal") == 0){
```

Comment:
See this comment, https://github.com/symengine/symengine.rb/pull/49#issuecomment-223286825
It's much better than comparing strings.